src/backoffice/settings/languages/workspace/language/language-workspace.context.ts

```diff
@@ -0,0 +1,72 @@
+import { UmbLanguageStore, UmbLanguageStoreItemType } from '../../language.store';
+import type { LanguageDetails } from '@umbraco-cms/models';
+import { UmbControllerHostInterface } from 'src/core/controller/controller-host.mixin';
+import { UniqueBehaviorSubject } from 'src/core/observable-api/unique-behavior-subject';
+import { UmbContextConsumerController } from 'src/core/context-api/consume/context-consumer.controller';
+import { UmbObserverController } from '@umbraco-cms/observable-api';
+
+const DefaultLanguageData: UmbLanguageStoreItemType = {
+	id: 0,
+	key: '',
+	name: '',
+	isoCode: '',
+	isDefault: false,
+	isMandatory: false,
+};
+
+export class UmbWorkspaceLanguageContext {
+	public host: UmbControllerHostInterface;
+
+	private _entityKey!: string;
+
+	private _data;
+	public readonly data;
+
+	private _store: UmbLanguageStore | null = null;
+	protected _storeObserver?: UmbObserverController<LanguageDetails>;
+
+	constructor(host: UmbControllerHostInterface, entityKey: string) {
+		this.host = host;
+		this._entityKey = entityKey;
+
+		this._data = new UniqueBehaviorSubject<LanguageDetails>(DefaultLanguageData);
+		this.data = this._data.asObservable();
+
+		new UmbContextConsumerController(host, 'umbLanguageStore', (_instance: UmbLanguageStore) => {
+			this._store = _instance;
+			if (!this._store) {
```

Comment:
Not sure I understand what this is trying to achieve. Are you unsure if you get the store here or something undefined instead? In theory, this callback should only be run if there is an actual store by that name.